File: goblobBase/src/main/java/com/cauchymop/goblob/view/InGameView.java
package com.cauchymop.goblob.view;


import com.cauchymop.goblob.viewmodel.InGameViewModel;

public interface InGameView {
  void setInGameModel(InGameViewModel inGameViewModel);
  void setInGameEventListener(InGameEventListener inGameEventListener);

  interface InGameEventListener extends GoBoardView.BoardEventListener {
    void onPass();
    void onDone();
  }
}

File: goblobBase/src/main/java/com/cauchymop/goblob/view/InGameView.kt
package com.cauchymop.goblob.view


import com.cauchymop.goblob.viewmodel.InGameViewModel

interface InGameView {
    fun setInGameModel(inGameViewModel: InGameViewModel)
    fun setInGameEventListener(inGameEventListener: InGameEventListener?)

    interface InGameEventListener : GoBoardView.BoardEventListener {
        fun onPass()
        fun onDone()
        fun onUndo()
        fun onRedo()
        fun onResign()
    }
}
